src/main/java/server/router/routes/RouteTemplate.java

```diff
@@ -0,0 +1,91 @@
+package server.router.routes;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import server.constants.Status;
+import server.request.Request;
+import server.response.Response;
+import server.response.ResponseBuilder;
+
+public abstract class RouteTemplate {
+  protected final Request request;
+  protected Status status;
+
+  protected RouteTemplate(Request request) {
+    this.request = request;
+  }
+
+  public abstract String path();
+
+  protected abstract List<String> methods();
+
+  protected abstract String resource();
+
+  public Response processRequest() {
+    return new ResponseBuilder()
+            .setStartLine(getStartLine())
+            .setHeaders(getHeaders())
+            .setBody(getBody())
+            .build();
+  }
+
+  protected String getStartLine() {
+    StringBuilder startLine = new StringBuilder();
+    startLine.append(request.protocol());
+    startLine.append(" ");
+
+    if (methods().contains(request.method())) {
+      status = Status.OK;
+    } else {
+      status = Status.NOT_ALLOWED;
+    }
+
+    startLine.append(status);
+
+    return startLine.toString();
+  }
+
+  private ArrayList<String> getHeaders() {
```

Comment:
You're using `ArrayList` and `List` over the project. What is the difference between them?

Reply:
I intended to only use `ArrayList` when I need a dynamic list that I will be adding elements to but I just looked and noticed some instances where I could be using a `List` for a collection with fixed length. I'm going to go through and update these

Reply:
updated all collections to be List type except for the ArrayList being used to collect headers for a response